src/wp-includes/script-loader.php

```diff
@@ -3027,3 +3027,522 @@ function wp_enqueue_block_style( $block_name, $args ) {
 	// Enqueue assets in the editor.
 	add_action( 'enqueue_block_assets', $callback );
 }
+
+/**
+ * Runs the theme.json webfonts handler.
+ *
+ * Using `WP_Theme_JSON_Resolver`, it gets the fonts defined
+ * in the `theme.json` for the current selection and style
+ * variations, validates the font-face properties, generates
+ * the '@font-face' style declarations, and then enqueues the
+ * styles for both the editor and front-end.
+ *
+ * Design Notes:
+ * This is not a public API, but rather an internal handler.
+ * A future public Webfonts API will replace this stopgap code.
+ *
+ * This code design is intentional.
+ *    a. It hides the inner-workings.
+ *    b. It does not expose API ins or outs for consumption.
+ *    c. It only works with a theme's `theme.json`.
+ *
+ * Why?
+ *    a. To avoid backwards-compatibility issues when
+ *       the Webfonts API is introduced in Core.
+ *    b. To make `fontFace` declarations in `theme.json` work.
+ *
+ * @link  https://github.com/WordPress/gutenberg/issues/40472
+ *
+ * @since 6.0.0
```

Comment:
```suggestion
 * @since 6.0.0
 * @access private
```

Should it be noted in the docblock that even with the design of the API it is possible to disable enqueueing webfonts in JSON files by running `remove_action( 'plugins_loaded', '_wp_theme_json_webfonts_handler' );`?